### ISSUE TITLE
Do not recommend extensions if language is set via auto detection

### ISSUE THREAD

**Issue by rebornix** [microsoft/vscode#208434]
Does this issue occur when all extensions are disabled?: Yes/No

* Create an untitled file and type `class A`
* It auto detects language C# and prompts

![image](https://github.com/microsoft/vscode/assets/876920/601e8981-00a4-4dbd-a188-3f5645a8df09)

IMHO we should not do any extension recommendation if the language is auto detected which can be wrong.

**Comment by TylerLeonhardt** [comment]
Not sure there's currently a way to soft set the language of the editor. This isn't possible, right @bpasero?

I'm trying to think of the right way for this to work.

**Comment by bpasero** [comment]
Isn't that the purpose of `hasLanguageSetExplicitly` in `BaseTextEditorModel`?

**Comment by TylerLeonhardt** [comment]
True. That's available. @sandy081 where would a good place be to check for this?

**Comment by sandy081** [comment]
Here: 

https://github.com/microsoft/vscode/blob/884c8a6566bcb8b6b12994fc21f175896f0efae7/src/vs/workbench/contrib/extensions/browser/fileBasedRecommendations.ts#L138

https://github.com/microsoft/vscode/blob/884c8a6566bcb8b6b12994fc21f175896f0efae7/src/vs/workbench/contrib/extensions/browser/fileBasedRecommendations.ts#L241
